saleor/graphql/payment/mutations/transaction/transaction_event_report.py

```diff
@@ -283,6 +288,94 @@ def clean_amount_value(
+        manager: "PluginsManager",
+        user: Optional["User"],
+        app: Optional[App],
+        previous_authorized_value: Decimal,
+        previous_charged_value: Decimal,
+        previous_refunded_value: Decimal,
+        related_granted_refund: Optional[order_models.OrderGrantedRefund],
+    ):
+        order = cast(order_models.Order, transaction.order)
+        order = updates_amounts_and_search_vector_for_order_with_lock(
+            order,
+            update_fields=[
+                "total_charged_amount",
+                "charge_status",
+                "updated_at",
+                "total_authorized_amount",
+                "authorize_status",
+                "search_vector",
+            ],
+        )
+        order_info = fetch_order_info(order)
+        order_transaction_updated(
+            order_info=order_info,
+            transaction_item=transaction,
+            manager=manager,
+            user=user,
+            app=app,
+            previous_authorized_value=previous_authorized_value,
+            previous_charged_value=previous_charged_value,
+            previous_refunded_value=previous_refunded_value,
+        )
+        if related_granted_refund:
+            calculate_order_granted_refund_status(related_granted_refund)
+
+    @classmethod
+    def update_order_or_checkout(
+        cls,
+        transaction: payment_models.TransactionItem,
+        manager: "PluginsManager",
+        user: Optional["User"],
+        app: Optional[App],
+        previous_authorized_value: Decimal,
+        previous_charged_value: Decimal,
+        previous_refunded_value: Decimal,
+        related_granted_refund: Optional[order_models.OrderGrantedRefund],
+    ):
+        if transaction.order_id:
+            cls.update_and_notify_order(
+                transaction,
+                manager,
+                user,
+                app,
+                previous_authorized_value,
+                previous_charged_value,
+                previous_refunded_value,
+                related_granted_refund,
+            )
+        if transaction.checkout_id:
+            with traced_atomic_transaction():
+                locked_checkout = (
+                    Checkout.objects.select_for_update()
+                    .filter(pk=transaction.checkout_id)
+                    .first()
+                )
+                transaction = payment_models.TransactionItem.objects.select_for_update(
+                    of=("self",)
+                ).get(pk=transaction.pk)
+                if transaction.checkout_id and locked_checkout:
+                    transaction_amounts_for_checkout_updated_without_sync_webhooks(
+                        transaction, locked_checkout, manager, user, app
+                    )
+                else:
+                    # If the checkout was deleted, we still want to update the order associated with the transaction.
+                    cls.update_and_notify_order(
```

Comment:
Are you aware, that you're changing the lock ordering for this case? 
In standard case, you have:
- move to `update_and_notify_order`
   - lock order

but here you have:
- locked_transaction
- move to `update_and_notify_order`
   - lock order